### PR TITLE
Replace Q promise library with native Promise (or with polyfill)

### PR DESCRIPTION
**Pull request by bankyadam**
The reason behind this PR is that somehow (did not research deeply) the Q lib is not working with Cypress E2E.

### CODE REVIEW

**Comment by BalazsAnkes** [comment]
This PR is awesome 🎉

File: lib/test-setup.spec.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ before(function() {
 });
 
 beforeEach(function() {
-  this.sandbox = sinon.sandbox.create();
+  this.sandbox = sinon.createSandbox();
```
Comment (ejgyurisan) [inline lib/test-setup.spec.js]: fyi https://sinonjs.org/guides/migrating-to-5.0#sinon-is-now-a-default-sandbox

Comment (bankyadam) [reply lib/test-setup.spec.js]: Changes landed in `master`: https://github.com/emartech/emarsys-integration-client-js/commit/37690ff51decdc244cede05866e17c6ce5c4e900

**Pull request merged**